### ISSUE TITLE
Bedrock Claude : [ERROR] Pipeline failed on step 2: No code found in the response

### ISSUE THREAD

**Issue by ferrymul7** [gventuri/pandas-ai#777]
### System Info

Hi Team, 

I have been running fine with pandasAI for the past 2-3 weeks. But suddenly yesterday, it gives me responses "Unfortunately, I was not able to answer your question, because of the following error:\n\nNo code found in the response\n'" ... I tried the same code, it also randomly responses with OpenAI with the same error ... It seems to me something is wrong with PandasAI, please have a look as below. I will also attached the full pandasai.log

Thanks, Ferry Mulyadi

Python version: 3.7 
pandasai version : 1.5.3

Code as below ********

#Import your dependencies
from hdbcli import dbapi
import pandas as pd
from pandasai import SmartDataframe
from langchain.llms import bedrock

#Initialize your connection
conn = dbapi.connect(
    address='xxx.',
    port='443',
    user='xxx',
    password='xxx',
    encrypt=True, 
    sslValidateCertificate=True
)
#If no errors, print connected
print('connected\n')

schema = "USER1"
tablename = "ALL_RESERVATIONS"
data=pd.read_sql(f'select * from {schema}.{tablename}',conn)
print(data)

# Instantiate a LLM
model_parameter = {"temperature": 0, "max_tokens_to_sample": 300}
langchain_llm = bedrock.Bedrock(model_id="anthropic.claude-v2", model_kwargs=model_parameter, region_name="us-west-2")
df = SmartDataframe(data, config={"llm": langchain_llm, "verbose": False})

df.chat("Which HOTEL has the most number of Nights ?")

**********
Print data as below
   RESNO     ARRIVAL  Nights          HOTEL    TITLE FIRST NAME LAST NAME
0      7  2020-04-12       3        Airport      Mrs      Susan     Baker
1      5  2019-03-14      10        Airport      Mrs      Sally     Smith
2      6  2020-04-12      18       Atlantic  Company       None  TOOLware
3      3  2020-11-14       4  Eighth Avenue  Company       None  Datasoft
4      9  2020-12-23      16   Indian Horse       Mr      Peter     Brown
5      4  2019-02-01       2     Long Beach       Mr     George      Howe
6      8  2020-09-01       2     Ocean Star       Mr     Antony   Jenkins
7      1  2020-12-24       3        Regency      Mrs      Jenny    Porter
8      2  2020-12-24      10        Regency       Mr      Peter     Brown
9     10  2020-11-14       3     River Boat  Company       None  TOOLware

Result as below
'Unfortunately, I was not able to answer your question, because of the following error:\n\nNo code found in the response\n'

### 🐛 Describe the bug

Unfortunately, I was not able to answer your question, because of the following error:\n\nNo code found in the response\n'

**********
pandasai.log

2023-11-24 02:53:11 [INFO] Question: Which HOTEL has the most number of Nights ?
2023-11-24 02:53:11 [INFO] Running PandasAI with langchain_amazon_bedrock LLM...
2023-11-24 02:53:11 [INFO] Prompt ID: 3c7fd5ca-ec8e-447e-87a3-e57b62bbf73e
2023-11-24 02:53:11 [INFO] Executing Step 0: CacheLookup
2023-11-24 02:53:11 [INFO] Executing Step 1: PromptGeneration
2023-11-24 02:53:11 [INFO] Using prompt: <dataframe>
dfs[0]:10x7
RESNO,ARRIVAL,Nights,HOTEL,TITLE,FIRST NAME,LAST NAME
7,2020-04-12,3,Airport,Mrs,Susan,Baker
6,2020-04-12,18,Atlantic,Company,,TOOLware
5,2019-03-14,10,Airport,Mrs,Sally,Smith
</dataframe>

Update this initial code:
```python
# TODO: import the required dependencies
import pandas as pd

"""
The variable `dfs: list[pd.DataFrame]` is already declared.
1. Prep: preprocessing/cleaning
2. Proc: data manipulation (group, filter, aggregate)
3. Analyze data
If the user requests to create a chart, utilize the Python matplotlib library to generate high-quality graphics that will be saved directly to a file.

Return a "result" variable dict:
- type (possible values "string", "number", "dataframe", "plot")
    - value (can be a string, a dataframe or the path of the plot, NOT a dictionary)
    Examples: 
        { "type": "string", "value": f"The highest salary is {highest_salary}." }
        or
        { "type": "number", "value": 125 }
        or
        { "type": "dataframe", "value": pd.DataFrame({...}) }
        or
        { "type": "plot", "value": "temp_chart.png" }
"""
```

Q: Which HOTEL has the most number of Nights ?
Return the full updated code:
2023-11-24 02:53:11 [INFO] Executing Step 2: CodeGenerator
2023-11-24 02:53:20 [ERROR] Pipeline failed on step 2: No code found in the response

[pandasai.log](https://github.com/gventuri/pandas-ai/files/13454754/pandasai.log)

**Comment by gventuri** [comment]
@ferrymul7 we've just released a new version (1.5.4) that seems to fix the issue for most of the LLMs.
Please, let me know if it works or I'll investigate further for your specific use case :)

**Comment by ferrymul7** [comment]
@gventuri , thank you for this fix. It works after my short validation. I am resolving this ticket. Thanks for responding fast.
